src/main/java/net/snowflake/client/core/arrow/TwoFieldStructToTimestampNTZConverter.java

```diff
@@ -139,4 +122,39 @@ public boolean toBoolean(int index) throws SFException {
         ErrorCode.INVALID_VALUE_CONVERT, logicalTypeStr,
         SnowflakeUtil.BOOLEAN_STR, val);
   }
+
+  public static Timestamp getTimestamp(
+      long epoch,
+      int fraction,
+      boolean fromToString,
+      TimeZone tz,
+      TimeZone sessionTimeZone,
+      boolean treatNTZasUTC,
+      boolean useSessionTimezone,
+      boolean honorClientTZForTimestampNTZ)
+      throws SFException {
+
+    if (ArrowResultUtil.isTimestampOverflow(epoch)) {
+      if (fromToString) {
+        throw new TimestampOperationNotAvailableException(epoch, fraction);
+      } else {
+        return null;
```

Comment:
It is ok to return null here. What that's mean?

Reply:
I removed it. It was present there when refactoring but after looking into it, I don't see any value in this. Removed this check, instead throwing `TimestampOperationNotAvailableException `